src/Microsoft.TemplateEngine.Abstractions/ICacheParameter.cs

```diff
@@ -1,11 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
 namespace Microsoft.TemplateEngine.Abstractions
 {
+    /// <summary>
+    /// Defines the representation of a parameter in template cache.
+    /// </summary>
     public interface ICacheParameter
     {
-        string DataType { get; }
+        /// <summary>
+        /// Gets the type of the parameter.
+        /// </summary>
+        string? DataType { get; }
+
+        /// <summary>
+        /// Gets the default value to be used if the user did not provide a value.
+        /// </summary>
+        string? DefaultValue { get; }
 
-        string DefaultValue { get; }
+        /// <summary>
+        /// Gets the friendly name of the parameter to be displayed to the user.
```

Comment:
Probably worth to mention that DisplayName and Description are localizable but not sure